protocol-units/da-sequencer/node/src/tests/client.rs

```diff
@@ -29,46 +38,46 @@ async fn test_client_should_successfully_write_batch() {
 	//need to wait the server is started before connecting
 	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
 
-	let tx = Transaction::test_only_new(b"test data".to_vec(), 1, 123);
-
-	let txs = FullNodeTxs::new(vec![tx]);
+	let connection_string = format!("http://127.0.0.1:{}", grpc_address.port());
+	let mut client = DaSequencerClient::try_connect(&connection_string)
+		.await
+		.expect("gRPC client connection failed.");
 
-	//sign batch
-	let batch_bytes = bcs::to_bytes(&txs).expect("Serialization failed");
-	let signature = sign_batch(&batch_bytes, &signing_key);
+	mock_write_new_batch(&mut client, &signing_key, verifying_key).await;
 
-	// Serialize full node batch into raw bytes
-	let serialized =
-		serialize_full_node_batch(verifying_key, signature.clone(), batch_bytes.clone());
-
-	//send the bacth using the grpc client
+	//verify the block produced contains the batch.
```

Comment:
Is this a `TODO`?

Reply:
No it's the next test step. I've implemented it in this PR.